src/Sylius/Bundle/AdminBundle/Controller/CustomerStatisticsController.php

```diff
@@ -0,0 +1,104 @@
+ */
+
+namespace Sylius\Bundle\AdminBundle\Controller;
+
+use Sylius\Component\Core\Customer\Statistics\CustomerStatisticsProviderInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\CustomerInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+
+/**
+ * @author Jan Góralski <jan.goralski@lakion.com>
+ */
+class CustomerStatisticsController
+{
+    /**
+     * @var CustomerStatisticsProviderInterface
+     */
+    private $statisticsProvider;
+
+    /**
+     * @var RepositoryInterface
+     */
+    private $channelRepository;
+
+    /**
+     * @var RepositoryInterface
+     */
+    private $customerRepository;
+
+    /**
+     * @var EngineInterface
+     */
+    private $templatingEngine;
+
+    /**
+     * @param CustomerStatisticsProviderInterface $statisticsProvider
+     * @param RepositoryInterface $channelRepository
+     * @param RepositoryInterface $customerRepository
+     * @param EngineInterface $templatingEngine
+     */
+    public function __construct(
+        CustomerStatisticsProviderInterface $statisticsProvider,
+        RepositoryInterface $channelRepository,
+        RepositoryInterface $customerRepository,
+        EngineInterface $templatingEngine
+    ) {
+        $this->statisticsProvider = $statisticsProvider;
+        $this->channelRepository = $channelRepository;
+        $this->customerRepository = $customerRepository;
+        $this->templatingEngine = $templatingEngine;
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     *
+     * @throws HttpException
+     */
+    public function renderStatisticsAction(Request $request)
+    {
+        $customerId = $request->query->get('customerId');
+
+        /** @var CustomerInterface $customer */
+        $customer = $this->customerRepository->find($customerId);
+        if (null === $customer) {
+            throw new HttpException(
+                Response::HTTP_BAD_REQUEST,
+                sprintf('Customer with id %s doesn\'t exist.', $customerId)
+            );
+        }
+
+        $allOrdersStatistics = $this->statisticsProvider->getCustomerAllOrdersCountStatistics($customer);
+        if (null === $allOrdersStatistics) {
+            return $this->templatingEngine->renderResponse(
+                '@SyliusAdmin/Customer/Show/Statistics/ordersStatistics.html.twig'
+            );
+        }
+
+        $channelsOrdersStatistics = array_map(function (ChannelInterface $channel) use ($customer) {
+            return $this->statisticsProvider->getCustomerOrdersInChannelStatistics($customer, $channel);
+        }, $this->channelRepository->findAll());
+
+        $channelsOrdersStatistics = array_filter($channelsOrdersStatistics);
```

Comment:
`$this->statisticsProvider->getCustomerOrdersInChannelStatistics(...)` cannot return `null` (at least that is what `CustomerStatisticsProviderInterface` says).